AdminPortal/src/pages/dashboard.tsx

```diff
@@ -1,6 +1,285 @@
-/**
- * Renders the {url}/dashboard page.
- */
-export default function Dashboard() {
-  return <div>{'This is the dashboard!'}</div>;
+import React, { useState, useEffect } from 'react';
+import styles from '@/styles/Dashboard.module.css';
+import Dropdown from 'react-bootstrap/Dropdown';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import data from '../data/data.json';
+import { Lato } from '@next/font/google';
+import { Sidebar } from '@/components/sidebar';
+import Select, { InputActionMeta } from 'react-select';
+
+interface DataItem {
+  id: number;
+  dateApplied: string;
+  name: string;
+  mobileNumber: string;
+  licenseID: string;
+  licensePlate: string;
+  isChecked: boolean;
+  category: String;
+}
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface ControlStyles {
+  [key: string]: unknown;
+}
+
+export default function App() {
+  const [activeTab, setActiveTab] = useState<string>('Needs Review');
+
+  const [selectAllClicked, setSelectAllClicked] = useState(
+    new Array(4).fill(false)
```

Comment:
What does the number 4 here represent? Can we switch it to something less hard coded?

Reply:
Here the number 4 represents the 4 tabs on the top of the page. The boolean array is used to track which one is the active tab at any given instant. I have changed this 4 to a variable here and everywhere else it occurs. Thanks for the feedback!